### ISSUE TITLE
Add a read-only mode for maintenance

### ISSUE THREAD

**Issue by olizilla** [web3-storage/web3.storage#452]
Update all api endpoints that mutate the database to check a feature flag for read-only mode, and have them return a descriptive error if read-only mode is true. It'd be nice to have the error point to the status page, and only enable read-only mode in conjunction with an announcement on the stataus page.

**Comment by atopal** [comment]
We'd also have to disable file uploads from the website, right?

**Comment by olizilla** [comment]
Yes, all uploads go via the api, so this would cover that too, and #453 would be the job to visualise this state.
